create_commits.py

```diff
@@ -0,0 +1,347 @@
+# pyright: strict
+import base64
+import subprocess
+import sys
+from typing import Optional, TypedDict
+import logging
+
+import requests
+
+
+class FileDeletion(TypedDict):
+    """
+    A type hint for the FileDeletion GraphQL input object, which is an input to the
+    FileChanges GraphQL input object.
+    https://docs.github.com/en/graphql/reference/input-objects#filedeletion
+    """
+
+    path: str
+
+
+class FileAddition(TypedDict):
+    """
+    A type hint for the FileAddition GraphQL input object, which is an input to the
+    FileChanges GraphQL input object.
+    https://docs.github.com/en/graphql/reference/input-objects#fileaddition
+    """
+
+    path: str
+    contents: str
+
+
+class FileChanges(TypedDict):
+    """
+    A type hint for the FileChanges GraphQL input object, which is an input to the
+    createCommitOnBranch mutation.
+    Read about the mutation here:
+    https://docs.github.com/en/graphql/reference/mutations#createcommitonbranch
+    And the file changes input object here:
+    https://docs.github.com/en/graphql/reference/input-objects#filechanges
+    """
+
+    additions: list[FileAddition]
+    deletions: list[FileDeletion]
+
+
+class CommitMessage(TypedDict):
+    """
+    A type hint for the CommitMessage GraphQL input object, which is an input to the
+    createCommitOnBranch mutation.
+    Read about the mutation here:
+    https://docs.github.com/en/graphql/reference/mutations#createcommitonbranch
+    And the commit message input object here:
+    https://docs.github.com/en/graphql/reference/input-objects#commitmessage
+    """
+
+    body: str
+    headline: str
+
+
+def get_file_changes_from_local_commit_hash(commit_hash: str) -> FileChanges:
+    """
+    Create a file changes object for a specific commit.
+
+    Args:
+        commit_hash (str): The hash of the commit.
+
+    Returns:
+        dict: A dictionary representing the FileChanges object.
+    """
+
+    # setting config variable advice.detachedHead to false to avoid verbose messages in the logs
+    subprocess.run(
+        ["git", "config", "advice.detachedHead", "false"],
+        capture_output=True,
```

Comment:
i'm guessing you just wanted to make this silent?
```suggestion
        stdout=subprocess.DEVNULL,
        stderr=subprocess.DEVNULL,
```